compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/descriptors/JvmDeclarationFactory.kt

```diff
@@ -148,7 +148,15 @@ class JvmDeclarationFactory(
             createStaticFunctionWithReceivers(
                 defaultImpls, name, interfaceFun,
                 dispatchReceiverType = parent.defaultType,
-                origin = JvmLoweredDeclarationOrigin.DEFAULT_IMPLS
+                // If `interfaceFun` is not a real implementation, then we're generating stubs in a descendant
+                // interface's DefaultImpls. For example,
+                //
+                //     class I1 { fun f() { ... } }
+                //     class I2 : I1
+                //
+                // is supposed to allow using `I2.DefaultImpls.f` as if it was inherited from `I1.DefaultImpls`.
+                // The classes are not actually related and `I2.DefaultImpls.f` is not a fake override but a bridge.
+                origin = interfaceFun.resolveFakeOverride()!!.origin
```

Comment:
Maybe we need new origin kind? Actually it would be source origin with new code?